src/pushsource/_impl/model/azure.py

```diff
@@ -0,0 +1,55 @@
+import re
+
+from frozenlist2 import frozenlist
+
+from .. import compat_attr as attr
+from .conv import in_, instance_of, optional_str, optional
+from .vms import VMIPushItem
+
+
+VHDGeneration = ["V1", "V2"]
+
+
+@attr.s()
+class VHDPushItem(VMIPushItem):
+    """A :class:`~pushsource.PushItem` representing a Virtual Hard Disk (or VHD).
+
+    For push items of this type, the :meth:`~pushsource.PushItem.src` attribute
+    refers to a file containing the VHD file..
+
+    This library does not verify that the referenced file is a valid
+    snapshot.
+    """
+
+    generation = attr.ib(type=str, default="V2", validator=optional(in_(VHDGeneration)))
```

Comment:
@JAVGan woudl you mind to explain why there is "optional"  validator?
IMHO it should be required in case of default value.
But maybe I do not understand something :-)

Reply:
I've set it as optional so the object can be created without passing the `generation` as argument and it should default to "V2".

Without this "optional" it would be required to pass the argument `generation` during the instantiation and we can't retrieve this value from koji directly

Reply:
> Without this "optional" it would be required to pass the argument generation during the instantiation and we can't retrieve this value from koji directly

would you mind to add such doc comment?

Reply:
I've added in the class description, PTAL again

Reply:
tyvm

Reply:
> I've set it as optional so the object can be created without passing the `generation` as argument and it should default to "V2".

This doesn't sound right (as a justification for 'optional'), because the above is already achieved by setting default="V2".

Including optional in the validator has the *added* effect that it is possible to create an object having `generation=None`, but this is not "without passing the generation as argument", this is explicitly passing an argument of value None.  So, do you want that to be possible/do you expect to do that or not?

If not sure then I'd suggest removing it (not allowing None), as it's always easier to widen the range of allowed values later, than it is to narrow them (which is backwards-incompatible).